### PR TITLE
Add neural network training and data processing

### PR DESCRIPTION
**Pull request by fksontini**
Implement a neural network training process with normalization, backpropagation, and model saving.

### CODE REVIEW

**Comment by Mistgun1** [comment]
a lot of mistakes, we don't accept ai slop
